src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILInterpreter.cs

```diff
@@ -573,6 +581,52 @@ private void InterpretStoreLocal(int index)
             _locals[index] = PopWithValidation();
         }
 
+        private void InterpretLoadArgument(int index)
+        {
+            TypeDesc argument = _method.Signature[index];
+            TypeFlags category = argument.Category;
+
+            switch (category)
+            {
+                case TypeFlags.Boolean:
+                case TypeFlags.Char:
+                case TypeFlags.SByte:
+                case TypeFlags.Byte:
+                case TypeFlags.Int16:
+                case TypeFlags.UInt16:
+                case TypeFlags.Int32:
+                case TypeFlags.UInt32:
+                    _stack.Push(StackItem.FromInt32(GetArgument<int>(index)));
+                    break;
+                case TypeFlags.Int64:
+                case TypeFlags.UInt64:
+                case TypeFlags.Enum:
+                    _stack.Push(StackItem.FromInt64(GetArgument<long>(index)));
```

Comment:
Enum should have its underlying type, not int64.

Reply:
Is the underlying type always `int32`? It's my understand that it can also be an `int64`

Reply:
The underlying enum type can be any primitive integer type: byte, short, int, ... .

Reply:
It should be something like this: https://github.com/dotnet/corert/blob/db8723b1787e18f0124e94c387542d0ef3aac128/src/ILCompiler.CppCodeGen/src/CppCodeGen/ILToCppImporter.cs#L259